File: app/src/main/java/com/bureng/robocoinx/contract/SplashContract.java
package com.bureng.robocoinx.contract;

import com.bureng.robocoinx.model.request.SignUpRequest;
import com.bureng.robocoinx.model.view.ProfileView;

public interface SplashContract {
    interface View{
        void goHome(ProfileView profileView);
        void loadFingerprint(SignUpRequest signupRequest);
        void showMessage(String message);
    }

    interface Presenter{
        boolean authorize();
        Object getResponse();
    }

}

File: app/src/main/java/com/bureng/robocoinx/contract/SplashContract.kt
package com.bureng.robocoinx.contract

import com.bureng.robocoinx.model.request.SignUpRequest
import com.bureng.robocoinx.model.view.ProfileView

interface SplashContract {
    interface View {
        fun goHome(profileView: ProfileView)
        fun loadFingerprint(signUpRequest: SignUpRequest)
        fun showMessage(message: String, type: Int)
        fun showProgressBar(rawLoading: Int?)
        fun hideProgressBar()
    }

    interface Presenter {
        fun authorize(): Boolean
        fun loadActivity()
    }
}